functionary/builder/utils.py

```diff
@@ -87,7 +235,9 @@ def initiate_build(
                 environment=environment, name=package_definition.get("name")
             )
         except Package.DoesNotExist:
-            pass
+            package_obj = create_package_from_definition(
```

Comment:
I think deleting the package is likely to be problematic (as with deleting anything really).  Imagine this scenario:

A package exists and is active.  A bunch of tasking has happened against it.  It is then deactivated.  Someone attempts to publish a new version and the build fails.  You go to clean up the package because it's inactive.  All of the tasks against any of the functions in that package are deleted with it, which was probably not the intention.

If you want to add the active flag now, I would say just leave the package behind and inactive if the build fails.  We can figure out the best way to let people clean those up manually later.

Reply:
I ended up adding two distinct variables to prepare for that use case: `completed` and `enabled`

`completed` is meant to indicate that the Package has been built successfully at least once, and as such, can be rendered as a Package on the UI. Open to suggestions about changing this name.

`enabled` is meant for enabling or disabling a `Package` and it's related Functions. This would be what gets updated manually by the user when they wish to enable/disable a `Package`

The reason for this distinction is to separate out the state of "this package should be visible to clients" and "this package is available for use by clients"

Reply:
I consolidated the two fields into just a `status` field. Although, I'm still not sure how I feel about those two fields being represented by one field, since the `PENDING` and `COMPLETED` fields are really only set once in a package's life cycle, and they are used by two separate entities

Reply:
What are the "two separate entities" you are referring to here?

Also, to your point about it being set once and then never again, that same point also means that if you have a flag for "completed" it's basically `True` almost always and just sits there at `True` and will never again be `False` under any circumstance.  So it kind of just feels like a waste of space.  With the `status` field, once you move past the `COMPLETED` state you are no longer wasting space storing the fact that this thing successfully built once upon a time.

Reply:
To be clear, the storage space on the flag is not the concern.  It's just the fact that with the flag, once you set it to True, it kind of loses all value.  It also creates a theoretical situation where you could have something marked as not complete, but active, which obviously doesn't make sense and then you have to put in logic to check for etc.  Seems like a bad time all around.

Reply:
The two entities are the Builder and the User. In my mind, the Builder needed a separate flag that it can manipulate to indicate that the package has been built, while the User should have their own flag for toggling whether the `Package` is enabled or disabled. Your point about the extra flag for the builder being a bit wasteful, I hadn't considered that, so I agree the single `status` field makes sense